audit-checklists/Arbitrum.md

```diff
@@ -0,0 +1,95 @@
+# Arbitrum security checklist
+
+## General
+
+### Block number
+
+Invoking `block.number` in a smart contract on Arbitrum will return **L1** block number at which sequencer received transaction, not an **L2** block number. Also worth noting, block number being returned is not necessarily the latest L1 block number. It is the latest synced block number and syncing process occurs approximately every minute. In that period there can be ~5 new L1 blocks. So values returned by `block.number` on Arbitrum do not increase continuously but in "jumps" of ~5 blocks.
+
+> Always check for incorrect use of `block.number` on L2s, especially if it is used to track time over short periods.
+
+Bug examples: [1](https://code4rena.com/reports/2022-12-tigris#m-15-_checkdelay-will-not-work-properly-for-arbitrum-or-optimism-due-to-blocknumber-), [2](https://solodit.cyfrin.io/issues/incorrect-use-of-l1-blocknumber-on-arbitrum-cantina-none-uniswap-pdf)
+
+## L1 to L2 messaging
+
+Retryable tickets are the canonical way to execute L1 to L2 messages in Arbitrum. User submits the message in the L1 Inbox and once submission is finalized message will be (asynchronously) executed on L2. If execution fails for any reason, like TX running out of gas or smart contract reverts, ticket stays in buffer and can be manually redeemed (thus it is "retryable"). We will describe some common pitfalls related to the usage of retryable tickets.
```

Comment:
User submits the message in the L1 Inbox - Could you paste a link to the Arbitrum repository at some commit where this occurs? 
E.g. [L1 inbox](https://github.com/OffchainLabs/nitro-contracts/blob/70d3670/src/bridge/ERC20Inbox.sol#L46)?

Reply:
I added the link to the function that's used to create retryable tickets. 
Function you linked `depositERC20`, or it's equivalent [depositEth](https://github.com/OffchainLabs/nitro-contracts/blob/780366a0c40caf694ed544a6a1d52c0de56573ba/src/bridge/Inbox.sol#L202) in the Inbox used for ETH-based chains, is not used for retryables tickets, but for s specific type of crosschain message which is used to bridge native assets. The difference compared to retryable ticket is that native asset bridging does not execute any transaction on the L2, but ArbOS node will read the message from L1 inbox and simply credit the balance to the L2 destination address. I'll try to find if there was any bugs related to different crosschain msg types and add new section about it